### PR TITLE
Feat/updatecardwithdetails

### PR DESCRIPTION
**Pull request by PhilippWeiss85**
Hi,
had kind of a struggle with the new function on the CardResult Component... 
Task that have been done:
- Added CardResults Component
- Added Styling
- Refactored initial states
- Added UpdateCardDetail function

### CODE REVIEW

**Review by Kheirah — CHANGES_REQUESTED** [review]
Good work!

There's one detail worth looking into: it's possible to save results without entering the outcome of the three individual matches. I merely select won/lost and I'm able to save the empty results.

Please validate your data by checking if the user actually input anything :)

File: components/CardDetail/CardDetail.js [hunk 1]
```diff
@@ -1,10 +1,24 @@
...
+        Player1: data.secondsetplayerone,
+        Player2: data.secondsetplayertwo,
+      },
+      {
+        Player1: data.thirdsetplayerone,
+        Player2: data.thirdsetplayertwo,
+      }
+    );
+    showMoreDetails();
   }
```
Comment (Kheirah) [inline components/CardDetail/CardDetail.js]: This is where you need to look into validating your data 👀 😄 

Check for input that is insensible and don't let it slip by :)

**Comment by PhilippWeiss85** [comment]
> Good work!
> 
> There's one detail worth looking into: it's possible to save results without entering the outcome of the three individual matches. I merely select won/lost and I'm able to save the empty results.
> 
> Please validate your data by checking if the user actually input anything :)

Hi @Kheirah 
thanks for the feedback. I updated the results component. 
A user have to enter at leat won or lost if he wants to save his game-details. Further it is optional for him to enter a score. Scores will only be rendered if the user enter at least one whole set. Otherwise the score container wont appear.

Please check again :)

**Comment by Kheirah** [comment]
Hey Phillip,
I can still enter only one outcome like this:
2022-10-25 13_08_13-https___capstone-project-24y66o3mh-philippweiss85 vercel app_gamelist

And then the score is misrepresented.
2022-10-25 13_08_21-https___capstone-project-24y66o3mh-philippweiss85 vercel app_gamelist

There's no way around validating your input, I'm afraid 😂

**Comment by PhilippWeiss85** [comment]
Hi @Kheirah 
added form validation. Please try again

**Review by Kheirah — APPROVED** [review]
All right, looking better now!

There's still onneee more thing 😄:
If I enter only the second match and leave the first and third match empty, the data is being saved and displayed **as if I entered the first match**. In other words, I can't tell that I entered the second match. Maybe you don't even want to allow entering the second or third match without having entered the first one already.

I'm going to approve your changes so far. But have another look at the situation I described above and try finding a solution for it before your next assessment 😀

**Comment by PhilippWeiss85** [comment]
> I'm going to approve your changes so far. But have another look at the situation I described above and try finding a solution for it before your next assessment 😀

Hi @Kheirah 
i have added another validation rule (not sure if this kind of code is best practice but it is working as intended) Thanks for approval

**Pull request merged**